shim/opentelemetry-opencensus-shim/src/opentelemetry/shim/opencensus/_shim_tracer.py

```diff
@@ -0,0 +1,90 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import wrapt
+from opencensus.trace.blank_span import BlankSpan
+from opencensus.trace.tracers.base import Tracer as BaseTracer
+
+from opentelemetry import context, trace
+from opentelemetry.shim.opencensus._shim_span import ShimSpan
+
+_logger = logging.getLogger(__name__)
+
+_SHIM_SPAN_KEY = context.create_key("opencensus-shim-span-key")
+
+
+def set_shim_span_in_context(
+    span: ShimSpan, ctx: context.Context
+) -> context.Context:
+    return context.set_value(_SHIM_SPAN_KEY, span, ctx)
+
+
+def get_shim_span_in_context() -> ShimSpan:
+    return context.get_value(_SHIM_SPAN_KEY)
+
+
+# pylint: disable=abstract-method
+class ShimTracer(wrapt.ObjectProxy):
+    def __init__(
+        self, wrapped: BaseTracer, *, otel_tracer: trace.Tracer
+    ) -> None:
+        super().__init__(wrapped)
+        self._self_otel_tracer = otel_tracer
```

Comment:
How come this can't be named `self._otel_tracer`?

Reply:
It causes the field to be saved on ShimTracer rather than on the wrapped object. See [here](https://wrapt.readthedocs.io/en/latest/wrappers.html#custom-object-proxies:~:text=If%20creating%20a%20custom%20proxy%20and%20it%20needs%20to%20keep%20attributes%20of%20its%20own%20which%20should%20not%20be%20saved%20through%20to%20the%20wrapped%20object%2C%20those%20attributes%20should%20be%20prefixed%20with%20_self_.).